lib/puppet/provider/ssl_pkey/openssl.rb

```diff
@@ -1,6 +1,10 @@
 require 'pathname'
 require 'openssl'
-Puppet::Type.type(:ssl_pkey).provide(:openssl) do
+require File.join(File.dirname(__FILE__), '..', '..', '..', 'puppet/provider/openssl')
```

Comment:
I only commented on the one, but it's a repeated pattern
```suggestion
require File.join(__dir__, '..', '..', '..', 'puppet/provider/openssl')
```